executor/pkg/executor/cl_executor.go

```diff
@@ -297,3 +281,67 @@ func (cle *ChainlinkExecutor) Validate() error {
 	}
 	return nil
 }
+
+// GetMessageStatus checks if a message should be executed and/or retried.
+// Returns (shouldRetry bool, shouldExecute bool, error) to indicate whether the message should be retried (added back to heap) and executed.
+func (cle *ChainlinkExecutor) GetMessageStatus(ctx context.Context, message protocol.Message, currentTime int64) (bool, bool, error) {
+	messageID, err := message.MessageID()
```

Comment:
I noticed that the `MessageID` returns an error because it re-encodes the message everytime which I find strange. If a message is constructed from the encoded bytes, they should be available (after trying to decode) and if a message is constructed from the fields, it can be encoded right then and there and included in the struct right?